apps/web/src/providers/ElectronOauthProvider.tsx

```diff
@@ -0,0 +1,24 @@
+import { SignMethod } from '@affine/workspace/affine/login';
+import type React from 'react';
+import { useEffect } from 'react';
+
+import { useAffineLogIn } from '../hooks/affine/use-affine-log-in';
+
+export const ElectronOauthProvider: React.FC<React.PropsWithChildren> = ({
```

Comment:
I think this hook can be embedded into `handleLogin` without introducing a new SignMethod.
in handleLogin:
- check if it is electron
- invoke apis.loginAffineCloud
- listen token callback (once)
- handle token callback

thus there is no need to have a new hook for it and all login workflow will be in `useAffineLogIn`